seacrowd/sea_datasets/filipino_hatespeech_tiktok/filipino_hatespeech_tiktok.py

```diff
@@ -0,0 +1,129 @@
+
+
+class FilipinoHateSpeechTikTok(datasets.GeneratorBasedBuilder):
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description="Filipino TikTok Hatespeech source schema",
+            schema="source",
+            subset_id="filipino_hatespeech_tiktok",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_text",
+            version=SEACROWD_VERSION,
+            description="Filipino TikTok Hatespeech Seacrowd schema",
+            schema="seacrowd_text",
+            subset_id="filipino_hatespeech_tiktok",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features({"text": datasets.Value("string"), "label": datasets.Value("string")})
+        elif self.config.schema == "seacrowd_text":
+            features = schemas.text.features(label_names=["0", "1"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        base_dir = _URL
+        data_files = {
+            "train": Path(dl_manager.download_and_extract(base_dir + "/train.csv")),
+            "test": Path(dl_manager.download_and_extract(base_dir + "/test.csv")),
+            "validation": Path(dl_manager.download_and_extract(base_dir + "/valid.csv")),
+        }
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_files["train"],
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": data_files["test"],
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": data_files["validation"],
+                    "split": "validation",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        df = pd.read_csv(filepath).reset_index()
+        id = 0
+        if self.config.schema == "source":
+            for row in df.itertuples():
+                ex = {
+                    "text": row.text,
+                    "label": str(int(row.label)) if not pd.isna(row.label) else "0",
```

Comment:
Is this behavior also used on their original dataset? Or is it being used only for convenience?

if yes, I'd like to suggest adding a reference link to it; else, you may skip it on iteration so that we don't alter the label/rule from original implementation

Reply:
Hi, I've updated it, thanks! :D